lib/storage/providers/LocalForage.js

```diff
@@ -7,39 +7,32 @@
 import localforage from 'localforage';
 import _ from 'underscore';
 import lodashMerge from 'lodash/merge';
+import SyncQueue from '../../SyncQueue';
 
 localforage.config({
     name: 'OnyxDB'
 });
 
-const writeQueue = [];
-let isQueueProcessing = false;
-
-/**
- * Writing very quickly to IndexedDB causes performance issues and can lock up the page and lead to jank.
- * So, we are slowing this process down by waiting until one write is complete before moving on
- * to the next.
- */
-function processNextWrite() {
-    if (isQueueProcessing || writeQueue.length === 0) {
-        return;
-    }
-
-    isQueueProcessing = true;
+const provider = {
+    /**
+     * Writing very quickly to IndexedDB causes performance issues and can lock up the page and lead to jank.
+     * So, we are slowing this process down by waiting until one write is complete before moving on
+     * to the next.
+     */
+    setItemQueue: new SyncQueue(({key, value}) => localforage.setItem(key, value)),
+    multiMergeQueue: new SyncQueue((pairs) => {
+        const tasks = _.map(pairs, ([key, partialValue]) => this.default.getItem(key)
+            .then((existingValue) => {
+                const newValue = _.isObject(existingValue)
+                    ? lodashMerge({}, existingValue, partialValue)
+                    : partialValue;
+                return this.default.setItem(key, newValue);
```

Comment:
Surprising to see `this` usages are still working

Reply:
Just realized we can use `localforage.setItem` directly now